### PR TITLE
imenu support

### PR DESCRIPTION
**Pull request by velkyel**

### CODE REVIEW

File: janet-mode.el [hunk 1]
```diff
@@ -86,6 +86,12 @@ the syntax table, so `forward-word' works as expected.")
 (defconst janet-constant-pattern
   (rx-to-string `(group symbol-start (group (or "true" "false" "nil")) symbol-end)))
 
+(defconst janet-imenu-generic-expression
+  '((nil
+     "^(\\(defn\\)*\\s-+(?\\(\\sw+\\)" 2)
```
Comment (ALSchwalm) [inline janet-mode.el]: I think I'd like this to be written in terms of `rx-to-string` like the rest of the regular expressions, just for clarity.

File: janet-mode.el [hunk 2]
```diff
@@ -402,6 +408,9 @@ STATE is the `parse-partial-sexp' state for that position."
   (setq-local comment-start-skip "#+ *")
   (setq-local comment-use-syntax t)
   (setq-local comment-end "")
+  (setq-local imenu-case-fold-search t)
+  (setq-local imenu-generic-expression janet-imenu-generic-expression)
+  (setq-local imenu-syntax-alist '(("+-*/.<>=?!$%_&~^" . "w")))
```
Comment (ALSchwalm) [inline janet-mode.el]: If the matches in `janet-imenu-generic-expression` are written like the other patterns, I don't think we should need to make changes to the syntax table for imenu.

**Review by ALSchwalm — CHANGES_REQUESTED** [review]
Thanks for the pull request! Just a couple of little things to address before I merge.

File: janet-mode.el (hunk 1 above)
Comment (ALSchwalm) [reply janet-mode.el]: Oh and for these I think we should probably also show private functions and macros (so `defn-` and `defmacro-`) as well.

Comment (velkyel) [reply janet-mode.el]: ok, I'll include defn- and defmacro-.  I have to learn rx-to-string first..

File: janet-mode.el (hunk 2 above)
Comment (velkyel) [reply janet-mode.el]: you are right, imenu-syntax-alist is unnecessary.

**Comment by pierre-rouleau** [comment]
@ ALSchwalm What's the status on this?  I'm starting to look at Janet and I'd like to get support going, including imenu support.  Not having imenu support breaks the prog-mode-hook.  Would it not be possible to allow this PR to go in and then modify it further?   I don't have much time but I might be able to help.

**Comment by ALSchwalm** [comment]
imenu support has now been added in 9e3254a0249d720d5fa5603f1f8c3ed0612695af, so I'm gonna go ahead and close this

**Comment by pierre-rouleau** [comment]
> imenu support has now been added in [9e3254a](https://github.com/ALSchwalm/janet-mode/commit/9e3254a0249d720d5fa5603f1f8c3ed0612695af), so I'm gonna go ahead and close this

Thanks!

**Pull request closed without merging**